akka-http/src/main/scala/akka/http/javadsl/server/Route.scala

```diff
@@ -50,9 +50,23 @@ trait Route {
    *    will be already be handled using the implicitly given [[RejectionHandler]] and [[ExceptionHandler]] (or
    *    the default handlers if none are given or can be found implicitly).
    *  - Consequently, no route alternatives will be tried that were combined with this route.
+   * @deprecated Use the variant without [[Materializer]]
    */
+  @Deprecated
   def seal(system: ActorSystem, materializer: Materializer): Route
 
+  /**
+   * Seals a route by wrapping it with default exception handling and rejection conversion.
+   *
+   * A sealed route has these properties:
+   *  - The result of the route will always be a complete response, i.e. the result of the future is a
+   *    ``Success(RouteResult.Complete(response))``, never a failed future and never a rejected route. These
+   *    will be already be handled using the implicitly given [[RejectionHandler]] and [[ExceptionHandler]] (or
+   *    the default handlers if none are given or can be found implicitly).
+   *  - Consequently, no route alternatives will be tried that were combined with this route.
+   */
+  def seal(system: ActorSystem): Route
```

Comment:
Is that method really needed?

Reply:
well, if the other one with an `ActorSystem` and `Materializer` was needed, this one is as well...
I kind of like this minimalist way to seal a route.
What's your idea? Just deprecate the one with `ActorSystem` and `Materializer`?

Reply:
Yep, I don't think either one is necessary.

Reply:
Leaving the `seal` variant with `RoutingSettings`, `ParserSettings`, `RejectionHandler`, `ExceptionHandler` as the only way to do it, right?

Reply:
Then, we'll might want to update this doc:
https://doc.akka.io/docs/akka-http/current/routing-dsl/routes.html?language=java#sealing-a-route

Reply:
However, I still like to be able to seal the route in a concise way.

Reply:
I yes right, can't we just remove the parameter?

Reply:
You mean remove without deprecating the old method?
If so, it's a non binary compatible change AFAIU

Reply:
No, actually deprecating is the right step. However, instead of introducing a new `seal(ActorSystem)`, we can directly create this one:

```scala
  override def seal(): Route = RouteAdapter {
    // TODO: scaladsl.server.Route.seal shouldn't require an implicit RoutingSettings when it can retrieve it itself
    akka.http.scaladsl.server.directives.BasicDirectives.extractSettings { implicit settings ⇒
      scaladsl.server.Route.seal(delegate)
    }
  }
```

Can you create a ticket to fix the scaladsl `Route.seal` not to require implicit parameters any more?

Reply:
Ah, yes, good idea. I'll change the `seal` to have no parameters.

I will also create the ticket you mention :+1:

Reply:
Cool, thanks a lot, @jlprat.